File: CP/code/intro/uva_11498.java
import java.util.Scanner;

class Main {
  public static void main(String[] args) {
    int K;

    Scanner in = new Scanner(System.in);
    while (true) {
      K = in.nextInt();
      if (K == 0)
        return;

      int N = in.nextInt();
      int M = in.nextInt();
      while (K-- > 0) {
        int resX = in.nextInt();
        int resY = in.nextInt();

        if (resX > N && resY > M) {
          System.out.printf("NE\n");
        } else if (resX > N && resY < M) {
          System.out.printf("SE\n");
        } else if (resX < N && resY > M) {
          System.out.printf("NO\n");
        } else if (resX < N && resY < M) {
          System.out.printf("SO\n");
        } else
          System.out.printf("divisa\n");
      }
    }
  }

}

File: CP/code/intro/uva_11498.cc
#include <cstdio>

using namespace std;

int main(int argc, char *argv[])
{
    int K;

    while (true) {
        scanf("%d", &K);
        if (K == 0)
            return 0;

        int N, M;
        int resX, resY;
        scanf("%d %d", &N, &M);
        while (K-- > 0) {
            scanf("%d %d", &resX, &resY);

            if (resX > N && resY > M) {
                printf("NE\n");
            } else if (resX > N && resY < M) {
                printf("SE\n");
            } else if (resX < N && resY > M) {
                printf("NO\n");
            } else if (resX < N && resY < M) {
                printf("SO\n");
            } else
                printf("divisa\n");
        }
    }
    return 0;
}
